### ISSUE TITLE
Soft hyphen and Non-breaking space invisible when Invisibles are activated (OS X)

### ISSUE THREAD

**Issue by bildlich** [atom/atom#10053]
The [Soft Hyphen](https://en.wikipedia.org/wiki/Soft_hyphen) is a useful character. It should be visible when Invisibles are turned on.

Atom 1.2.4 on OS X 10.11.1

**Reproduction Steps:**

1. In Atom, turn on invisibles through the Shift+CMD+P menu and place the cursor where you want to insert the soft hyphen
2. Open the OS character window with Ctrl+CMD+Space
3. Search for the Soft Hyphen and double click on it to insert into Atom
4. Nothing happens
5. Open the file in another editor, e.g. TextWrangler. The hyphen is there.

**Comment by bildlich** [comment]
The same goes for the [Non-breaking space](https://en.wikipedia.org/wiki/Non-breaking_space). It can be reproduced easier. On OS X, type ALT+Space to create a Non-breaking space. Here is a comparison of how Atom (left) and TextWrangler (right) treat a Non-Breaking Space with invisibles turned on.

![screenshot 2015-12-14 14 50 54](https://cloud.githubusercontent.com/assets/501617/11782609/34c35e1c-a272-11e5-98a2-2d7e0b4fcc95.png)

**Comment by lee-dohm** [comment]
This is blocked by https://github.com/atom/atom/issues/2572.

**Comment by chharvey** [comment]
@bildlich yes! The fact that non-breaking spaces (U+00A0) and normal spaces (U+0020) look the same, even with invisibles turned on, is slightly frustrating.

@lee-dohm  what does the **blocked** label mean, and what do you mean by "blocked by [another issue]"?

**Comment by 50Wliu** [comment]
@chharvey It means that #2572 needs to be fixed first before this issue can.

**Comment by mik01aj** [comment]
+1, I just ran into this. Soft hyphen is a special case, I guess. It's displayed zero-width, but I can still put my cursor before or after it.

**Comment by chharvey** [comment]
#2572 has been closed in favor of #12157

**Comment by lee-dohm** [comment]
This is replaced now by #12157. You may want to subscribe there for updates.
